docs/proposals/scheduledjob.md

```diff
@@ -216,6 +208,144 @@ In the above example:
+1. there are now two jobs for the same start time.
+
+We can solve this in several ways:
+
+1. with three-phase protocol, e.g.:
+  1. controller creates a "suspended" job.
+  1. controller writes writes an annotation in the SJ saying that it created a job for this time.
+  1. controller unsuspends that job.
+1. by picking a deterministic name, so that at most one object create can succeed.
+
+#### Ensuring one Pod
+
+Job object does not currently have a way to ask for this.
+Even if it did, controller is not written to support it.
+Same problem as above.
+
+#### Ensuring one container invocation per Pod
+
+Kubelet is not written to ensure at-most-one-container-start per pod.
+
+#### Decision
+
+This is too hard to do for the alpha version.  We will await user
+feedback to see if the "at most once" property is needed in the beta version.
+
+This is awkward but possible for a containerized application ensure on it own, as it needs
+to know what ScheduledJob name and Start Time it is from, and then record the attempt
+in a shared storage system.   We should ensure it could extract this data from its annotations
+using the downward API.
+
+## Name of Jobs
+
+A ScheduledJob creates one Job at each time when a Job should run.
+Since there may be concurrent jobs, and since we might want to keep failed
+non-overlapping Jobs around as a debugging record, each Job created by the same ScheduledJob
+needs a distinct name.
+
+To make the Jobs from the same ScheduledJob distinct, we could use a random string,
+in the way that pods have a `generateName`.  For example, a scheduledJob named `nightly-earnings-report`
+in namespace `ns1` might create a job `nightly-earnings-report-3m4d3`, and later create
+a job called `nightly-earnings-report-6k7ts`.  This is consistent with pods, but
+does not give the user much information.
+
+Alternatively, we can use time as a uniqifier.  For example, the same scheduledJob could
+create a job called `nightly-earnings-report-2016-May-19`.  
+However, for Jobs that run more than once per day, we would need to represent
+time as well as date.  Standard date formats (e.g. RFC 3339) use colons for time.
+Kubernetes names cannot include time.  Using a non-standard date format without colons
+will annoy some users.   
+
+Also, date strings are much longer than random suffixes, which means that
+the pods will also have long names, and that we are more likely to exceed the
+253 character name limit when combining the scheduled-job name,
+the time suffix, and pod random suffix.
+
+One option would be to compute a hash of the nominal start time of the job,
+and use that as a suffix.  This would not provide the user with an indication
+of the start time, but it would prevent creation of the same execution 
+by two instances (replicated or restarting) of the controller process.
+
+We chose to use the random suffix approach for the alpha version.
+We may use the time-hash approach later if we decide to offer the
+at-most-once property.
```

Comment:
Because, although the short generated postfix is consistent with pods/rc in other parts of the system. In those places a user rarely cares about underlying objects. Whereas here, as mentioned, he might be.